src/blocked.c

```diff
@@ -634,16 +634,29 @@ void unblockClientWaitingData(client *c) {
     }
 }
 
+int getBlockedTypeByType(int type) {
+    switch (type) {
+    case OBJ_LIST: return BLOCKED_LIST;
+    case OBJ_ZSET: return BLOCKED_ZSET;
+    case OBJ_MODULE: return BLOCKED_MODULE;
+    case OBJ_STREAM: return BLOCKED_STREAM;
+    default: return BLOCKED_NONE;
+    }
+}
+
 /* If the specified key has clients blocked waiting for list pushes, this
  * function will put the key reference into the server.ready_keys list.
  * Note that db->ready_keys is a hash table that allows us to avoid putting
  * the same key again and again in the list in case of multiple pushes
  * made by a script or in the context of MULTI/EXEC.
  *
  * The list will be finally processed by handleClientsBlockedOnKeys() */
-void signalKeyAsReady(redisDb *db, robj *key) {
+void signalKeyAsReady(redisDb *db, robj *key, int btype) {
     readyList *rl;
 
+    /* If this type is not blocked, just return */
+    if (!server.blocked_clients_by_type[btype]) return;
```

Comment:
A minor suggestion. If this function were to take the object type instead of the blocked type and we do the conversion in this function, it would move more of the blocking code in blocked.c. You could omit the getBlockedTypeByType from server.h as well. It might add some extra overhead, but I think will help with maintainability.

Reply:
@madolson Thanks, already updated.